### ISSUE TITLE
One Segment Battleground setting not respected in Battlegrounds

### ISSUE THREAD

**Issue by hazzal** [Tercioo/Details-Damage-Meter#210]
### **Before Start!**
**Make sure the bug also happen when Details! is the only addon enabled.**

**Game and Details! version**
Use /details to grab these two.
R8501  v9.0.5

**Describe the bug**
- Steps to Reproduce (what did you do to make the bug happen):
Enable **_One Segment Battleground_** option in settings. Enter a rated BG, enter combat multiple times and check your segments.
- Result (what happens when you follow the steps above):
You get multiple segments, one for for each fight inside the battleground.
- Expected (what you think the expected behavior would be when following the steps):
That you get one segment for one battleground when you have the setting enabled.
- When the bug started (a date, wow path release, details update, etc)?
Started happening with one the latest update, had a friend with the same problem too.
- Additional information:
I haven't tested this in Random BGs yet but I expect the same behavior happens there. Will update with a comment if I test in an unrated BG.

**Lua Error**
Post here lua errors, if you aren't getting any, use */console scriptErrors 1*

**Screenshots**
Post images of the bug if necessary.

**Combat Log**
Include a link to a combatlog file if the error is about not detecting a skill, usable item or item proc.
The log can be as simple as hitting a mob with white damage and using the item/skill.

**Other Information**
If necessary, add more info here.

**Comment by hazzal** [comment]
Just tested this in a random BG and I get same result as in rated BGs. According to a friend the issue doesn't appear to happen in R8406. I think I have narrowed down what causes the issue and will do brief testing to see if I can fix it.

**Comment by stevebeck89** [comment]
I'm also seeing this issue.  Been happening for the past day or so.

```
[00:33:58] Details!: DETAILS! VERSION: R8501
[00:33:58] Details!: GAME VERSION: v9.0.5
```

**Comment by hazzal** [comment]
Latest commit seems to have fixed this issue e8a0380153fd775985c9fd867d5711b4df18e068
